csharp/src/Apache.Arrow/Arrays/BinaryArray.cs

```diff
@@ -171,6 +171,7 @@ public TBuilder Clear()
         public ReadOnlySpan<byte> Values => ValueBuffer.Span.CastTo<byte>();
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [Obsolete]
```

Comment:
It's probably best to add a message to the `[Obsolete]` attribute telling users what they should use instead.